dask/array/rechunk.py

```diff
@@ -651,3 +651,31 @@ def format_plan(plan):
     [(3*[10], 2*[15]), ([30], 3*[10])]
     """
     return [format_chunks(c) for c in plan]
+
+
+def _get_shape_helper(a):
+    s = np.asarray(a.shape, dtype=int)
+    return s[len(s) * (None,) + (slice(None),)]
+
+
+def _get_all_chunk_shapes(a):
+    return a.map_blocks(
+        _get_shape_helper,
+        dtype=int,
+        chunks=tuple(len(c) * (1,) for c in a.chunks) + ((a.ndim,),),
+        new_axis=a.ndim,
+    )
+
+
+def _get_chunks(a):
+    cs = _get_all_chunk_shapes(a)
```

Comment:
Style nit: I prefer to avoid indirection like this if it's not necessary. Reasoning here: https://matthewrocklin.com/blog/work/2019/06/23/avoid-indirection

Reply:
I might just move all of this code into the `compute_metadata` method, placing the `_get_chunks` somewhere else in core.py.  I don't think that there is a strong need to keep it elsewhere.

Reply:
@stsievert can I ask that we remove the function `_get_chunk_shapes`, and just put those lines of code inside of the `compute_chunk_sizes` method?

Reply:
Oh, thanks for the reminder. I've pushed an update.